src/connectors/onchain_events/mod.rs

```diff
@@ -249,6 +255,25 @@ impl Subscriber {
             }
             Some(events) => events.push(event.clone()),
         }
+        if block_number as u64 > self.latest_block_in_db() {
+            match ingest_state::onchain_events::put_state(
```

Comment:
This is at the wrong level. `add_onchain_event` gets call multiple times per block. This should be at least one level higher, in process_log. But it would be even better to do this one per batch when syncing historical events in bulk, and then once per block when syncing live events.